refactor(login): migrate to ts, Chakra

LoneRifle opened the pull request:
## Problem and Solution

As part of an overall effort to depend more on Chakra,
migrate away from using manual styling and JavaScript to Chakra theming and TypeScript

## Tests
- Ensure login screen looks the same

LoneRifle committed:
refactor(login): migrate to ts, Chakra

LinHuiqing commented on client/src/pages/Login/Login.component.tsx:
```diff
@@ -1,23 +1,30 @@
         <Spacer h={['64px', '64px', '84px']} />
-        <h2>AskGov</h2>
-        <p>Answers from the Singapore Government</p>
+        <Box as="h2" textStyle="display-2" mb="8px">
+          AskGov
+        </Box>
```
`as="h2"` does not do anything. Also `Text` component might be a better component to use as this is just text. Something like:
```ts
<Text textStyle="display-2" mb="8px">
  AskGov
</Text>
```

LoneRifle replied:
So something like this? ok.
```suggestion
        <Text textStyle="display-2" mb="8px">
          AskGov
        </Text>
```

LinHuiqing replied:
yup 👍

LinHuiqing commented on client/src/pages/Login/Login.component.tsx:
```diff
@@ -1,23 +1,30 @@
+        mt="48px"
+        px={{ base: '36px', xs: '48px' }}
+        mx="auto"
+        maxW="calc(320px + 48px * 2)"
```
`"calc(...)"` should be changed to a constant since calculation would always be the same

LoneRifle replied:
I feel that this is clearer, as it shows what we are looking for, ie, "320 pixels of width, with additional provisions for padding"

LinHuiqing commented on client/src/pages/Login/Login.component.tsx:
```diff
@@ -1,23 +1,30 @@
+        px={{ base: '36px', xs: '48px' }}
+        mx="auto"
+        maxW="calc(320px + 48px * 2)"
+        className="login-page"
```
```suggestion

```
Unnecessary line^

LoneRifle replied:
Let's keep that for now - we might want to further refactor the theming to use a separate file

LoneRifle committed:
refactor(login): use Text, not Box, for heading

We avoid use of heading as its styling interferes with
our own, and we do not have time to test the unstyled
variant

Co-authored-by: Lin Huiqing <37061143+LinHuiqing@users.noreply.github.com>
